monegros-restaurant-dianamonegrodiaz/src/monegros/restaurant/entities/Order.java

```diff
@@ -0,0 +1,107 @@
+package monegros.restaurant.entities;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+
+public class Order {
+    private Long id;
+    private Chef chef;
+    private Instant transactionDate;
+    private Client client;
+    private Cashier cashier;
+    private List<OrderDetail> orderDetails;
+    private List<IngredientsOfOrderDetail> ingredientsOfOrder;
```

Comment:
Why does `IngredientsOfOrderDetail` extends from `OrderDetail` and also you have a list of `OrderDetail` under the same `Order`?

Reply:
Because the list of ingredients of a product has a name (Ex: Pizza, Pizza is list of ingredients), a price, an id, the order that it belongs to, etc.
The List<OrderDetail> orderDetails was there beforehand and it's used to specify the elements of an specific Order. It does not contains or refers to the ingredients of each element of that Order.

Reply:
> Because the list of ingredients of a product has a name (Ex: Pizza, Pizza is list of ingredients), a price, an id, the order that it belongs to, etc.
> The List orderDetails was there beforehand and it's used to specify the elements of an specific Order. It does not contains or refers to the ingredients of each element of that Order.

Doesn't make sense to me since an `OrderDetail` has a product and if a product is an instance of a FinalProduct then you could get the ingredient of that `FinaProduct` if you'd want.

Is there any other reason I'm not getting on why you are creating this entity?

Either way talking about business logic, why would you want to attach ingedients to an order of client? I mean, does a client need to know the ingredients of his order? isn't this a problem to expose your ingredients outside to people? If we talk about reality another restaurant could copy your food by knowing the ingredients you used

Reply:
I was just going to use the ingredients of the order internally in the project, it wasn't supposed to be presented to the client. I will fix it.

Reply:
> I was just going to use the ingredients of the order internally in the project, it wasn't supposed to be presented to the client. I will fix it.

But according to the logic... isn't the order suppose to be given to the client? My question is, why something that is supposed to do something has to do something different. If the `Order` is supposed to be given to the client in order to know what he bought and how he spent on it, why do we have to carry about internal processes on it?